### PR TITLE
docs: example of image occurrence comparison in Typescript/Javascript

### PR DESCRIPTION
**Pull request by MagicPoulp**
## Proposed changes

document for JS the image detection based on forums discussion.
Duplicated PR to try the CLA signing again.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x ] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [x ] Lint and unit tests pass locally with my changes
- [x ] I have added tests that prove my fix is effective or that my feature works
- [x ] I have added necessary documentation (if appropriate)
- [x ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by MagicPoulp** [comment]
exact content as in previous PR.

File: docs/en/writing-running-appium/image-comparison.md [hunk 1]
```diff
@@ -97,6 +99,69 @@ File.open('find_result_visual.png', 'wb') { |f| f<< Base64.decode64(find_result_
...
+     The Javascript client driver webdriverio does not support (in January 2020) the "-image" strategy implemented in the Appium server. You will have more power and understanding while using openCV directly. Since the appium server is in Javascript, you can do all it does with opencv in your test suite.
+
+     The testing framework mocha can be run with typescript to have async/await.
+     You need to run mocha with those options in the right order and with the associated packages installed:
+     NODE_PATH=/path/to/nodejs/lig/node_modules TS_NODE_PROJECT=config/tsconfig_test.json --require ts-node/register --require tsconfig-paths/register
+     You will also need to make a basic config/tsconfig_test.json
+     Note that paths in tsconfig.json does not support absolute paths. Hence, you cannot move the NODE_PATH there.
+  */
+  import * as path from 'path';
+  const cv = require(path.join(process.env.NODE_PATH, '/opencv4nodejs'));
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/image-comparison.md]: nit: `path.join` automatically adds slashes

File: docs/en/writing-running-appium/image-comparison.md [hunk 2]
```diff
@@ -97,6 +99,69 @@ File.open('find_result_visual.png', 'wb') { |f| f<< Base64.decode64(find_result_
...
+  const isImagePresent = async () => {
+    /// Take screenshot and read the image
+    const screenImagePath = './appium_screenshot1.png';
+    await driver.saveScreenshot(screenImagePath)
+    const likedImagePath = './occurrence1.png';
+
+    // Load images
+    const originalMatPromise = cv.imreadAsync(screenImagePath);
+    const waldoMatPromise = cv.imreadAsync(likedImagePath);
+    const [originalMat, waldoMat] = await Promise.all([originalMatPromise, waldoMatPromise]);
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/image-comparison.md]: 👍

**Review by mykola-mokhnach — APPROVED** [review]

File: docs/en/writing-running-appium/image-comparison.md (hunk 1 above)
Comment (MagicPoulp) [reply docs/en/writing-running-appium/image-comparison.md]: fixed

**Comment by MagicPoulp** [comment]
Ready to merge.

**Review by mykola-mokhnach — APPROVED** [review]

**Comment by MagicPoulp** [comment]
Is there anything I can do to help merge this?

**Comment by MagicPoulp** [comment]
Note, I just added this comment at the end:
"    // then you know if the image was found by comparing the rectangle with a reference rectangle.
    // the structure minMax contains the property maxVal that gives the quality of the match
    // 1 is prefect match, but you may get .999. If you extract an image from the screenshot manually,
    // you will get an image that matches.
"

**Pull request merged**